### ISSUE TITLE
atan2._eval_evalf is missing a return keyword

### ISSUE THREAD

**Issue by smite** [sympy/sympy#17461]
`sympy.functions.atan2._eval_evalf` is missing a return keyword, and consequently `atan2` does not work with symbolic arguments that redefine their `_eval_evalf` method.

See https://groups.google.com/forum/#!msg/sympy/fpRIf9c79zw/taAssuspAgAJ

Solution: Add the return keyword.

```
import sympy
import sympy.functions as syf

class A(sympy.Symbol):
    def _eval_evalf(self, prec):
        return sympy.Float(5.0)

x = A('X', real=True)
y = A('Y', real=True)
print(syf.atan2(y, x).evalf())  # prints "atan2(Y, X)" instead of "0.785398163397448"
```

**Comment by oscarbenjamin** [comment]
Thanks. Feel free to send a PR but otherwise I've marked this as "good first issue" so that a new contributor will hopefully find it.

**Comment by oscargus** [comment]
I added a fix and test for it in #17278

**Comment by oscargus** [comment]
Btw, would it make sense to have an else-clause rewriting it to log if the arguments are not extended_real? That would handle sub-classing with imaginary evalf-values, but also lead to that symbolic values are rewritten. Maybe one can actually to the evalf for the log-rewrite and if it is not a number, just return the original expression?
